modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptAngularClientCodegen.java

```diff
@@ -247,8 +250,55 @@ public void postProcessParameter(CodegenParameter parameter) {
                 }
             }
 
-            // Convert path to TypeScript template string, applying URI encoding
-            op.path = op.path.replaceAll("\\{(.*?)\\}", "\\$\\{encodeURIComponent(String($1))\\}");
+            // Generate the regex objects needed to parse the path properly.
+            Pattern pathPattern = Pattern.compile("\\{(.*?)\\}");
+            Matcher pathMatcher = pathPattern.matcher(op.path);
+
+            // Prep a string buffer where we're going to set up our new version of the string.
+            StringBuffer pathBuffer = new StringBuffer();
+
+            // Iterate through the matches for the regex.
+            // This should be anything that was contained inside curly-braces.
+            while(pathMatcher.find()) {
+                // Get a substring of our match.
+                String templateSection = op.path.substring(pathMatcher.start(), pathMatcher.end());
```

Comment:
I would choose a simpler implementation like e.g. https://stackoverflow.com/a/34229320

Reply:
Hmm. Looking at it now, I agree. It's been a long while since I've coded in Java, so after seeing the original code used regex, I just did the same in case it needed to be more complicated later. That said, it's definitely not the most efficient. 

I'll mess with this is a bit, I should be able to do this with much simpler string manipulation, crawling through 1 character at a time. I should only need one iteration through the string.